Core/HLE/sceKernelModule.cpp

```diff
@@ -1269,17 +1257,44 @@ static Module *__KernelLoadELFFromPtr(const u8 *ptr, size_t elfSize, u32 loadAdd
 	if (textSection == -1) {
 		module->textStart = reader.GetVaddr();
 		module->textEnd = firstImportStubAddr - 4;
+	}
 
-		if (!module->isFake) {
-#if !defined(MOBILE_DEVICE)
-			bool gotSymbols = reader.LoadSymbols();
-			MIPSAnalyst::ScanForFunctions(module->textStart, module->textEnd, !gotSymbols);
-#else
-			if (g_Config.bFuncReplacements) {
-				bool gotSymbols = reader.LoadSymbols();
-				MIPSAnalyst::ScanForFunctions(module->textStart, module->textEnd, !gotSymbols);
-			}
+	if (!module->isFake) {
+		bool scan = true;
+#if defined(MOBILE_DEVICE)
+		scan = g_Config.bFuncReplacements;
 #endif
+
+		bool gotSymbols = scan && reader.LoadSymbols();
+		std::vector<SectionID> codeSections = reader.GetCodeSections();
+		for (SectionID id : codeSections) {
+			u32 start = reader.GetSectionAddr(id);
+			// Note: scan end is inclusive.
+			u32 end = start + reader.GetSectionSize(id) - 4;
+
+			if (start < module->textStart)
+				module->textStart = start;
+			if (end > module->textEnd)
+				module->textEnd = end;
+
+			if (scan)
+				MIPSAnalyst::ScanForFunctions(start, end, !gotSymbols);
+		}
+
+		// Some games don't have any sections at all.
```

Comment:
Really? Homebrew or real games?

Reply:
Well, I only know of one so far - Frontier Gate Boost (demo).  I'm not certain it's unmodified, though...

-[Unknown]